docs/reference/src/main/asciidoc/configure.asciidoc

```diff
@@ -273,6 +273,37 @@ The functionality is implemented as a built-in portable extension processing all
 |`org.jboss.weld.bootstrap.vetoTypesWithoutBeanDefiningAnnotation` || A regular expression. If a non-empty string, then all annotated types whose `AnnotatedType#getJavaClass().getName()` matches this pattern are vetoed if not annotated with a bean defining annotation.
 |=======================================================================
 
+[[remove-unused-beans]]
+==== Memory consumption optimization - removing unused beans
+
+CDI applications consist of user-defined beans implementing the business logic, but also beans coming from libraries (e.g. DeltaSpike) and integrations (e.g. various Java EE technologies - JSF, Bean Validation, Batch).
+Most applications actually use only a small part of the beans provided by libraries and integrations.
+Still, Weld has to retain the metadata for all beans in memory!
+What a waste.
+If <<allow-optimized-cleanup>> is allowed Weld can now remove the _unused_ beans after bootstrap.
+
+An _unused_ bean:
```

Comment:
While this definition is precise, it can be hard to imagine, at first glance, what will actually be removed by this optimization. I would add some more general statement (saying what _is_ unused bean) and then follow up with this definition?
Rough idea - `For instance, if you run Probe and list all the beans in deployed EE application, you will almost certainly notice some of them coming from 3rd party libraries which you never use (such as <insert name of batch bean?>). Those are candidates for _unused beans_ as you can be sure you are not using them. To be precise, an _unused bean_:`

Reply:
We could add this as a NOTE.

Reply:
Could be, I am just trying to make it a bit exemplary. To show user that the problem of hanging unused beans in memory is very likely affecting their application as well.
NOTE would do just fine I guess.